### PR TITLE
[nrf temphack] pm: Add partition manager support to 'settings'

### PR DESCRIPTION
**Pull request by SebastianBoe**
Add partition manager support to the settings subsystem.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

**Comment by pdunaj** [comment]
With the change above settings still do not initialize. I checked that problem is error in one of the function calls initializing settings that causes panic (explicit panic on error).
The same problem is in other areas in zephyr - in our case `zephyr/subsys/settings/src/settings_fcb.c` also has invalid id.

File: subsys/settings/src/settings_init.c [hunk 1]
```diff
@@ -59,7 +71,7 @@ static void settings_init_fcb(void)
 	int rc;
 	const struct flash_area *fap;
 
-	rc = flash_area_get_sectors(DT_FLASH_AREA_STORAGE_ID, &cnt,
+	rc = flash_area_get_sectors(FLASH_AREA_STORAGE_ID, &cnt,
```
Comment (pdunaj) [inline subsys/settings/src/settings_init.c]: Also needed in `zephyr/subsys/settings/src/settings_fcb.c` at line 45.

**Comment by SebastianBoe** [comment]
> The same problem is in other areas in zephyr - in our case zephyr/subsys/settings/src/settings_fcb.c also has invalid id.

Fixed.

File: subsys/settings/src/settings_init.c (hunk 1 above)
Comment (SebastianBoe) [reply subsys/settings/src/settings_init.c]: Fixed.

File: subsys/settings/src/settings_fcb.c [hunk 2]
```diff
@@ -13,6 +13,18 @@
 #include "settings/settings_fcb.h"
 #include "settings_priv.h"
 
+#if USE_PARTITION_MANAGER
```
Comment (nvlsianpu) [inline subsys/settings/src/settings_fcb.c]: just why this is not incorporated into a header file?

Comment (SebastianBoe) [reply subsys/settings/src/settings_fcb.c]: It should be.

I'll add that to the partition manager TODO list.

Comment (nvlsianpu) [reply subsys/settings/src/settings_fcb.c]: :D

**Pull request merged**